### ISSUE TITLE
Handling multiple changes in multiple Content Types

### ISSUE THREAD

**Issue by afonsojramos** [udamir/api-smart-diff#9]
I suppose that the duplicate removal is not really fixable, and that makes sense. But the description of it is always `text/plain`, even  if the paths are correct.

<img width="477" alt="image" src="https://github.com/udamir/api-smart-diff/assets/19473034/ba4e8dab-5918-4d43-ad6e-558bcb6b2ad4">

**Before**
```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "Content Type Demo",
    "description": "Demo of content type.",
    "version": "2.0"
  },
  "paths": {
    "/accounts/{accountId}": {
      "get": {
        "tags": [
          "Accounts"
        ],
        "summary": "Returns standard account details.",
        "operationId": "GetAccount",
        "responses": {
          "400": {
            "description": "Bad Request",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              }
            }
          },
          "404": {
            "description": "Not Found",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "415": {
            "description": "Unsupported Media Type",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "500": {
            "description": "Server Error",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "502": {
            "description": "Bad Gateway",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Customer": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "description": "Unique Customer Identifier (GUID or Customer Number)",
            "nullable": true,
            "example": "56344fa1-4f24-41b9-b0ae-fd40b998ef2c"
          },
          "type": {
            "allOf": [
              {
                "$ref": "#/components/schemas/CustomerType"
              }
            ],
            "description": "Customer type (Primary, Additional - In Future)",
            "example": "Primary"
          }
        },
        "additionalProperties": false,
        "description": "Customer"
      },
      "CustomerType": {
        "enum": [
          "Primary"
        ],
        "type": "string"
      },
      "GetAccountResponse": {
        "type": "object",
        "properties": {
          "customers": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/Customer"
            },
            "description": "List of Customers"
          },
          "createdAt": {
            "type": "string",
            "description": "Account creation Date (+time)",
            "format": "date-time",
            "nullable": true,
            "example": "2023-11-21T00:00:00.0000000+00:00"
          }
        },
        "additionalProperties": false,
        "description": "Get Account Response"
      },
      "ProblemDetails": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string",
            "description": "A URI reference [RFC3986] that identifies the problem type. This specification encourages that, when\r\ndereferenced, it provide human-readable documentation for the problem type\r\n(e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be\r\n\"about:blank\".",
            "nullable": true
          },
          "title": {
            "type": "string",
            "description": "A short, human-readable summary of the problem type.It SHOULD NOT change from occurrence to occurrence\r\nof the problem, except for purposes of localization(e.g., using proactive content negotiation;\r\nsee[RFC7231], Section 3.4).",
            "nullable": true
          },
          "status": {
            "type": "integer",
            "description": "The HTTP status code([RFC7231], Section 6) generated by the origin server for this occurrence of the problem.",
            "format": "int32",
            "nullable": true
          },
          "detail": {
            "type": "string",
            "description": "A human-readable explanation specific to this occurrence of the problem.",
            "nullable": true
          },
          "instance": {
            "type": "string",
            "description": "A URI reference that identifies the specific occurrence of the problem.It may or may not yield further information if dereferenced.",
            "nullable": true
          }
        },
        "additionalProperties": {},
        "description": "A machine-readable format for specifying errors in HTTP API responses based on https://tools.ietf.org/html/rfc7807."
      }
    }
  }
}
```

**After**
```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "Content Type Demo",
    "description": "Demo of content type.",
    "version": "2.0"
  },
  "paths": {
    "/accounts/{accountId}": {
      "get": {
        "tags": [
          "Accounts"
        ],
        "summary": "Returns standard account details.",
        "operationId": "GetAccount",
        "responses": {
          "400": {
            "description": "Bad Request",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/GetAccountResponse"
                }
              }
            }
          },
          "404": {
            "description": "Not Found",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "415": {
            "description": "Unsupported Media Type",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "500": {
            "description": "Server Error",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          },
          "502": {
            "description": "Bad Gateway",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ProblemDetails"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Customer": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "description": "Unique Customer Identifier (GUID or Customer Number)",
            "nullable": true,
            "example": "56344fa1-4f24-41b9-b0ae-fd40b998ef2c"
          },
          "type": {
            "allOf": [
              {
                "$ref": "#/components/schemas/CustomerType"
              }
            ],
            "description": "Customer type (Primary, Additional - In Future)",
            "example": "Primary"
          }
        },
        "additionalProperties": false,
        "description": "Customer"
      },
      "CustomerType": {
        "enum": [
          "Primary"
        ],
        "type": "string"
      },
      "GetAccountResponse": {
        "type": "object",
        "properties": {
          "customers": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/Customer"
            },
            "description": "List of Customers"
          },
          "createdAt": {
            "type": "string",
            "description": "Account creation Date (+time)",
            "format": "date-time",
            "example": "2023-11-21T00:00:00.0000000+00:00"
          }
        },
        "additionalProperties": false,
        "description": "Get Account Response"
      },
      "ProblemDetails": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string",
            "description": "A URI reference [RFC3986] that identifies the problem type. This specification encourages that, when\r\ndereferenced, it provide human-readable documentation for the problem type\r\n(e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be\r\n\"about:blank\".",
            "nullable": true
          },
          "title": {
            "type": "string",
            "description": "A short, human-readable summary of the problem type.It SHOULD NOT change from occurrence to occurrence\r\nof the problem, except for purposes of localization(e.g., using proactive content negotiation;\r\nsee[RFC7231], Section 3.4).",
            "nullable": true
          },
          "status": {
            "type": "integer",
            "description": "The HTTP status code([RFC7231], Section 6) generated by the origin server for this occurrence of the problem.",
            "format": "int32",
            "nullable": true
          },
          "detail": {
            "type": "string",
            "description": "A human-readable explanation specific to this occurrence of the problem.",
            "nullable": true
          },
          "instance": {
            "type": "string",
            "description": "A URI reference that identifies the specific occurrence of the problem.It may or may not yield further information if dereferenced.",
            "nullable": true
          }
        },
        "additionalProperties": {},
        "description": "A machine-readable format for specifying errors in HTTP API responses based on https://tools.ietf.org/html/rfc7807."
      }
    }
  }
}
```

**Diff**
```json
[
  {
    "path": [
      "paths",
      "/accounts/{accountId}",
      "get",
      "responses",
      "200",
      "content",
      "text/plain",
      "schema",
      "properties",
      "createdAt",
      "nullable"
    ],
    "before": true,
    "action": "remove",
    "type": "breaking",
    "description": "[Removed] possbile nullable value from `createdAt` in Response 200 content (text/plain)"
  },
  {
    "path": [
      "paths",
      "/accounts/{accountId}",
      "get",
      "responses",
      "200",
      "content",
      "application/json",
      "schema",
      "properties",
      "createdAt",
      "nullable"
    ],
    "before": true,
    "action": "remove",
    "type": "breaking",
    "description": "[Removed] possbile nullable value from `createdAt` in Response 200 content (text/plain)"
  },
  {
    "path": [
      "paths",
      "/accounts/{accountId}",
      "get",
      "responses",
      "200",
      "content",
      "text/json",
      "schema",
      "properties",
      "createdAt",
      "nullable"
    ],
    "before": true,
    "action": "remove",
    "type": "breaking",
    "description": "[Removed] possbile nullable value from `createdAt` in Response 200 content (text/plain)"
  },
  {
    "path": [
      "components",
      "schemas",
      "GetAccountResponse",
      "properties",
      "createdAt",
      "nullable"
    ],
    "before": true,
    "action": "remove",
    "type": "non-breaking"
  }
]
```

**Comment by udamir** [comment]
It is not duplication - it's changes in different parts of api, so it works as designed. 
The issue with the wrong content type in the annotation is due to caching the comparison results to optimize the comparison logic. I will fix it in next patch.

**Comment by udamir** [comment]
fixed in api-smart-diff@1.0.5

**Comment by afonsojramos** [comment]
Working, thanks!
